tests/sparse_weights_test.py

```diff
@@ -0,0 +1,63 @@
+#  Numenta Platform for Intelligent Computing (NuPIC)
+#  Copyright (C) 2019, Numenta, Inc.  Unless you have an agreement
+#  with Numenta, Inc., for a separate license for this software code, the
+#  following terms and conditions apply:
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Affero Public License version 3 as
+#  published by the Free Software Foundation.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the GNU Affero Public License for more details.
+#
+#  You should have received a copy of the GNU Affero Public License
+#  along with this program.  If not, see http://www.gnu.org/licenses.
+#
+#  http://numenta.org/licenses/
+#
+
+import unittest
+
+import torch
+
+from nupic.torch.modules import SparseWeights, SparseWeights2d
+
+
+class TestSparseWeights(unittest.TestCase):
+
+    def test_sparse_weights_1d(self):
+        in_features, out_features = 784, 10
+        with torch.no_grad():
+            for percent_on in [0.1, 0.5, 0.9]:
+                linear = torch.nn.Linear(in_features=in_features,
+                                         out_features=out_features)
+                sparse = SparseWeights(linear, percent_on)
+                nonzeros = torch.nonzero(sparse.module.weight, as_tuple=True)[0]
+                counts = torch.unique(nonzeros, return_counts=True)[1]
+
+                # Expected activity per unit
+                expected = [round(in_features * percent_on)] * out_features
```

Comment:
This should check that each out_feature has `round(in_features * percent_on)` non-zero weights.

Reply:
That is exactly what it is being checked.
The `counts` tensor contains the count of nonzeros for each `out_features`. 
In the case where `in_features = 784` and `out_features=10` the tensor `counts` will contain the following:
```
[78, 78, 78, 78, 78, 78, 78, 78, 78, 78]
```

Reply:
Ah ok, thanks.